brave/brave-core
# [Uplift] - 1.65.x - Update SolanaWeb3 to 1.91.7

## Summary
- Updates Solana Web3 from 1.58.x to 1.91.x (major bump, not minor bump).


Uplift of: https://github.com/brave/brave-core/pull/23202
Resolves https://github.com/brave/brave-browser/issues/37664



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Visit https://codesandbox.io/s/github/darkdh/solana-provider-test
2. Tap on the `Open In New Window` button on the top right side of the page
3. <img width="34" alt="image" src="https://github.com/brave/brave-core/assets/1530031/ed3c8091-e583-44cc-be22-6c4425c06e0c">
4. `Connect` your wallet.
5. Go to Wallet Settings: 
6. <img width="287" alt="image" src="https://github.com/brave/brave-core/assets/1530031/f2d08abc-aadd-4079-8eca-c912f78409df">
7. Go to Wallet Networks: 
8. <img width="492" alt="image" src="https://github.com/brave/brave-core/assets/1530031/9fbcc6be-903d-49a4-b220-b50da8232076">
9. Enable Test Networks: Solana Dev-Net: 
10. <img width="435" alt="image" src="https://github.com/brave/brave-core/assets/1530031/c23b6fde-adeb-4416-9072-c8d1e582a294">
11. Make sure your account is on Solana Dev-Net Network: 
12. <img width="280" alt="image" src="https://github.com/brave/brave-core/assets/1530031/19ac10f9-6814-4d60-b334-bee6439e8e7b">
13. On the main screen, tap on `Deposit`: 
14. <img width="281" alt="image" src="https://github.com/brave/brave-core/assets/1530031/328cb52e-d11f-43e7-980f-57f385c080e4">
15. From the All Networks drop down, select Solana DevNet.
16. Copy your wallet address on the next screen.
17. Visit `https://faucet.solana.com`
18. Paste your wallet address, select 5 SOL as the amount, and click "Confirm Airdrop".
19. Go back to your first tab which the URL is `https://pwgoom.csb.app/`
20. Select `Sign and Send Transaction (v0)`.
21. Confirm the transaction: 
22. <img width="242" alt="image" src="https://github.com/brave/brave-core/assets/1530031/e2cf1960-323b-494c-b74b-f4979693c9a1">
23. Success! Everything is working.



## Screenshots

<img width="1353" alt="image" src="https://github.com/brave/brave-core/assets/1530031/8da63a04-8ef8-4dd4-9c5b-c48a76b8b70d">
